ports/vcpkg-cmake-get-vars/cmake_get_vars/CMakeLists.txt

```
@@ -101,20 +102,21 @@ if(APPLE)
         string(APPEND EXTRA_FLAGS " -arch ${arch}")
     endforeach()
     if(CMAKE_OSX_DEPLOYMENT_TARGET)
-        if(CMAKE_SYSTEM_NAME STREQUAL "iOS")
-            string(APPEND EXTRA_FLAGS " -mios-version-min=${CMAKE_OSX_DEPLOYMENT_TARGET}")
-        elseif(CMAKE_SYSTEM_NAME STREQUAL "Darwin")
-            string(APPEND EXTRA_FLAGS " -mmacosx-version-min=${CMAKE_OSX_DEPLOYMENT_TARGET}")
-        endif()
+        list(GET VCPKG_LANGUAGES 0 lang)
+        string(APPEND EXTRA_FLAGS " ${CMAKE_${lang}_OSX_DEPLOYMENT_TARGET_FLAG}${CMAKE_OSX_DEPLOYMENT_TARGET}")
+        unset(lang)
     endif()
 endif()
-if(CMAKE_CXX_COMPILER_TARGET)
-    if(CMAKE_CXX_COMPILER_ID STREQUAL "Clang")
-        string(APPEND EXTRA_FLAGS " \"${CMAKE_CXX_COMPILE_OPTIONS_TARGET}${CMAKE_CXX_COMPILER_TARGET}\"")
-    elseif(CMAKE_CXX_COMPILE_OPTIONS_TARGET)
-        string(APPEND EXTRA_FLAGS " ${CMAKE_CXX_COMPILE_OPTIONS_TARGET} \"${CMAKE_CXX_COMPILER_TARGET}\"")
+set(target_param "")
+foreach(lang IN_LIST VCPKG_LANGUAGES)

```

```suggestion
list(GET VCPKG_LANGUAGES 0 lang)
```

This loop should also be replaced

1. it's only taking one `target_param`
2. `foreach` doesn't work with `IN_LIST` :see_no_evil:

---

https://github.com/Neumann-A/vcpkg/pull/12